### DISCUSSION TITLE
Basic Auth?

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10801] [Q&A]
I spun up a aws-node-http-api-project/ and I'm utterly confused how to implement simply basic Auth. 

Do I need to make a custom authorizer?? https://medium.com/@Da_vidgf/http-basic-auth-with-api-gateway-and-serverless-5ae14ad0a270

I'm particularly bewildering since when I use:

```
const BasicAuth = require('express-basic-auth');
const Serverless = require('serverless-http');
```

It happens without any complications! But I want to achieve the same user experience without using a plugin/package

**Comment by mnapoli** [comment]
Hi,

What you did in code works well, as you noticed. That's a valid option.

The other option, if you don't want to do it in code, is to do it via API Gateway. In that case you will have to use the Authorizer features of API Gateway.

It's a choice and both options are valid.

**Reply by kaihendry** [reply]
Does the "Authorizer" feature mean I have to implement say a separate function as a Lambda authorizer?

**Reply by mnapoli** [reply]
I'm guessing so yes: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-access-control.html

**Reply by kaihendry** [reply]
I'm struggling to find a simply basic auth implementation in serverless without https://github.com/svdgraaf/serverless-basic-authentication
